ampd/src/lib.rs

```diff
@@ -255,6 +265,7 @@ where
                 completed_height.increment(),
             )),
         };
+        // TODO: pass block_height_monitor to handler
```

Comment:
Just pass the `Arc<AtomicU64>` over, instead of the entire monitor. If you wanna make it nicer than just a `Arc<AtomicU64>`, you can create a wrapper around it.